server/pages/index.js

```diff
@@ -376,20 +378,44 @@ module.exports = function() {
 		app.get( '/themes/:theme_slug', function( req, res ) {
 			const context = getDefaultContext( req );
 
-			if ( config.isEnabled( 'server-side-rendering' ) ) {
+			const renderThemeSheet = function( theme ) {
 				const store = createReduxStore();
+				store.dispatch( {
+					type: ActionTypes.RECEIVE_THEME_DETAILS,
+					themeId: theme.id,
+					themeName: theme.name,
+					themeAuthor: theme.author,
+					themeScreenshot: theme.screenshot,
+				} );
 
 				store.dispatch( setSection( 'themes', { hasSidebar: false, isFullScreen: true } ) );
-				context.initialReduxState = pick( store.getState(), 'ui' );
+				context.initialReduxState = pick( store.getState(), 'ui', 'themes' );
 
 				Object.assign( context, render( (
 					<ReduxProvider store={ store }>
 						<LayoutLoggedOutDesign store={ store } routeName={ 'themes' } match={ { theme_slug: req.params.theme_slug } } />
 					</ReduxProvider>
 				) ) );
-			}
+				res.render( 'index.jade', context );
+			};
 
-			res.render( 'index.jade', context );
+			if ( config.isEnabled( 'server-side-rendering' ) ) {
+				let themeData = themeDetails.get( req.params.theme_slug );
+				if ( ! themeData ) {
+					wpcom.undocumented().themeDetails( req.params.theme_slug, ( error, data ) => {
```

Comment:
> Yeah, was thinking about this, or maybe even using the api response header, which may let us leverage some existing libraries to handle it?

Yep. I wonder whether we sent ETags or what the Expires header is like.

> Once we receive new api data, how do we clear the renderToString cache?

You mean how do we know what key to clear? Maybe we key the rts cache using something like `dataTimestamp+lang`?